pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -263,17 +272,42 @@ func (r *reconciler) handleMirrorImage(ctx context.Context, targetImageRef strin
 	return nil
 }
 
-func checkAllBuildsSuccessful(builds map[string]*buildv1.Build) bool {
-	for _, build := range builds {
+func (r *reconciler) listBuilds(ctx context.Context, mabcName string) (*buildv1.BuildList, error) {
+	builds := buildv1.BuildList{}
+	requirement, err := labels.NewRequirement(v1.MultiArchBuildConfigNameLabel, selection.Equals, []string{mabcName})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create requirement: %w", err)
+	}
+	listOpts := ctrlruntimeclient.ListOptions{LabelSelector: labels.NewSelector().Add(*requirement)}
+	if err := r.client.List(ctx, &builds, &listOpts); err != nil {
+		return nil, fmt.Errorf("failed to list builds: %w", err)
+	}
+	return &builds, nil
+}
+
+// missingArchitecture returns a set of architectures which no builds have been created for
```

Comment:
This shouldn't be the case. When we will try to create the builds, if we won't create builds for all architectures we should fail.

Reply:
Should we fail and mark the `mabc` as failed?

Reply:
Yep.